endpoints/openrtb2/auction.go

```diff
@@ -1277,25 +1277,73 @@ func (deps *endpointDeps) validateUser(req *openrtb_ext.RequestWrapper, aliases
 	}
 
 	// Check Universal User ID
-	for eidIndex, eid := range req.User.EIDs {
-		if eid.Source == "" {
-			return append(errL, fmt.Errorf("request.user.eids[%d] missing required field: \"source\"", eidIndex))
-		}
+	if len(req.User.EIDs) > 0 {
+
+		validEids, eidErrors := validateEIDs(req.User.EIDs)
 
-		if len(eid.UIDs) == 0 {
-			return append(errL, fmt.Errorf("request.user.eids[%d].uids must contain at least one element or be undefined", eidIndex))
+		if len(eidErrors) > 0 {
+			errL = append(errL, eidErrors...)
 		}
+		for eidIndex, eid := range validEids {
+			if eid.Source == "" {
+				return append(errL, fmt.Errorf("request.user.eids[%d] missing required field: \"source\"", eidIndex))
+			}
 
-		for uidIndex, uid := range eid.UIDs {
-			if uid.ID == "" {
-				return append(errL, fmt.Errorf("request.user.eids[%d].uids[%d] missing required field: \"id\"", eidIndex, uidIndex))
+			if len(eid.UIDs) == 0 {
+				return append(errL, fmt.Errorf("request.user.eids[%d].uids must contain at least one element or be undefined", eidIndex))
+			}
+
+			for uidIndex, uid := range eid.UIDs {
+				if uid.ID == "" {
+					return append(errL, fmt.Errorf("request.user.eids[%d].uids[%d] missing required field: \"id\"", eidIndex, uidIndex))
+				}
```

Comment:
At this time we will not have empty `did.ID`, correct? there is a check for this already in line 1334.

Reply:
Also do we need to set "clean" EIDs back to `req.User.EIDs`?